src/main/java/EntryPoint/service/UserService.java

```diff
@@ -0,0 +1,103 @@
+package EntryPoint.service;
+
+import EntryPoint.dto.AccessTokenResponseDTO;
+import EntryPoint.dto.LoginResponseDTO;
+import EntryPoint.dto.UserDTO;
+import EntryPoint.dto.UserProfileDTO;
+import EntryPoint.exception.GlobalExceptionHandler.*;
+import EntryPoint.repository.UserRepository;
+import EntryPoint.utils.JwtUtil;
+import jakarta.mail.MessagingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import EntryPoint.model.User;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@Service
+@RequiredArgsConstructor
+
+public class UserService {
```

Comment:
As the class works for authentication thats why class name should be AuthenticationService

Reply:
resolved